include/aikido/planner/World.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef AIKIDO_PLANNER_WORLD_HPP_
+#define AIKIDO_PLANNER_WORLD_HPP_
+
+#include <string>
+#include <dart/dart.hpp>
+
+namespace aikido {
+namespace planner {
+
+class World
+{
+public:
+  /// Construct a kinematic World.
+  World(const std::string& name);
+
+  virtual ~World() = default;
+
+  /// Create a clone of this World. All Skeletons will be copied over.
+  std::shared_ptr<World> clone(const std::string& name) const;
```

Comment:
Nit: Need docstring for the parameter.

Also, I like you used `newName` in the source file for this parameter. It would be good to use the same name here too.